packages/foundation/src/createComponent.tsx

```diff
@@ -52,17 +52,18 @@ export function createComponent<
   ) => {
     const settings: ICustomizationProps<TViewProps, TTokens, TStyleSet> = _getCustomizations(
       options.displayName,
+      // eslint-disable-next-line react-hooks/rules-of-hooks
       React.useContext(CustomizerContext),
       options.fields,
     );
 
-    const useState = options.state;
+    const stateReducer = options.state;
```

Comment:
Renamed to avoid name collision with `useState` hook.

Reply:
Good idea, I'm going to borrow this for #14097 instead of disabling the warning.